docs/api/app.md

```diff
@@ -501,6 +501,41 @@ gets emitted.
 **Note:** Extra command line arguments might be added by Chromium,
 such as `--original-process-start-time`.
 
+### Event: 'notification-activation' _macOS_ _Windows_
+
+Returns:
+
+* `event` Event
+* `appUserModelId` string - contains App module ID
+* `invokedArgs` string - contains arguments(in case of any) for toast button or data property in case of non-persistent notification
+* `dataCount` integer - contains amount of token pairs present inputData string (not null, default value 0 - which means abcense of inputData)
+* `inputData` string - valid JSON-string which contains result of User input into toast Reply fields. Empty string in case of absence of User input (i.e for case dataCount == 0)
```

Comment:
The JSON parsed in `//base` is incredibly performant, given the array will either be empty `[]` or it will contain data and the user will want to run `JSON.parse` on it themselves I don't think the difference in performance will be noticeable / impactful in any way.  We should just make the nicer API and expose the parsed structure

Reply:
I think that some misunderstanding is present
We have next 

` Activate(_In_ LPCWSTR appUserModelId,
           _In_ LPCWSTR invokedArgs,
           _In_reads_(dataCount) const NOTIFICATION_USER_INPUT_DATA* data,
           ULONG dataCount) override {
    // packing user input structs into stream
    std::wstringstream stm;
    // SAP-15908 refine notification-activation for cold start
    if (dataCount) {
      stm << L"[";  // json array open brace
      std::for_each(data, data + dataCount,
                    [&](const NOTIFICATION_USER_INPUT_DATA& item) {
                      stm << item
                          // avoid problem with last delimeter
                          << std::wstring((static_cast<ULONG>(&item - data +
                                                              1) == dataCount)
                                              ? L""
                                              : L",");
                    });
      stm << L"]";  // json array close brace
    }`
    
To exclude consideration of the code base we have next scheme:
if (dataCount) -> inputData is JSON
if (dataCount == 0) this JSON array is empty

On the app module side to have dataCount we can immediatly provide prediction should we use JSON parser or not. JSON is necessary to integrate data for pressed button and in that case JSON is extremely comfortable form of serialization. 
We do not want to parse nothing data into C++ we are packing it for top level JS app module - that is idea

Reply:
Regarding to Your remarks about on first look not professional JSON forming by hands : we know format of data, we have only limited set of data variants I do not see necessity of usage some code from base instead of overridden ostream operator
`wostream& operator<<(wostream& out, const NOTIFICATION_USER_INPUT_DATA& data) {
  out << L"{";  // open brace for key-value pair
  out << L"\"" << data.Key << L"\"";
  out << L":";  // json delimeter (i.e colon) between key value
  out << L"\"" << data.Value << L"\"";
  out << L"}";  // close brace for key-value pair
  return out;
}` 
Please consider this code ax extremely high level of code optimization with purpose to  have maximal performance for huge sets of Notifications.